### PR TITLE
Openemr fix telehealth signup locale

### PR DESCRIPTION
**Pull request by adunsulag**
Fixed up the verbiage and some problems with the telehealth live payment signup process. It was popping up an alert box with the paypal subscriber id and expected the patient to copy and remember that value for signup.  This is most likely what was causing the multiple signup problems that users of the module were dealing with.

Broke the signup into three steps and provide progress status to the users as they signup.  Better instructions on how to move to the next step are included in each process.

### CODE REVIEW

File: interface/modules/custom_modules/oe-module-comlink-telehealth/welcome.php [hunk 1]
```diff
@@ -13,6 +13,15 @@
...
 require_once dirname(__FILE__, 4) . "/globals.php";
 
+use Comlink\OpenEMR\Modules\TeleHealthModule\Bootstrap;
+use Comlink\OpenEMR\Modules\TeleHealthModule\TelehealthGlobalConfig;
+
+$kernel = $GLOBALS['kernel'];
+$bootstrap = new Bootstrap($kernel->getEventDispatcher(), $kernel);
+$globalConfig = $bootstrap->getGlobalConfig();
+$subscriptionId = $globalConfig->getGlobalSetting(TelehealthGlobalConfig::COMLINK_TELEHEALTH_PAYMENT_SUBSCRIPTION_ID) ?? '';
+$isCoreConfigured = $globalConfig->isTelehealthCoreSettingsConfigured() === true;
```
Comment (sjpadgett) [inline interface/modules/custom_modules/oe-module-comlink-telehealth/welcome.php:23]: Very nice.

File: interface/modules/custom_modules/oe-module-comlink-telehealth/src/TelehealthGlobalConfig.php [hunk 2]
```diff
@@ -430,11 +443,27 @@ public function renderFooterBox($fldid, $fldarray)
...
+            ,'isLocaleConfigured' => $isLocaleConfigured
         ];
 
         return $this->twig->render("comlink/admin/telehealth_footer_box.html.twig", $dataArray);
     }
 
+    private function isLocaleConfigured()
+    {
+        // timezone is not set in the $GLOBALS array oddly, not sure why, check against the database
+        $record = QueryUtils::fetchRecords("SELECT gl_name, gl_index, gl_value FROM globals WHERE gl_name=?", [self::LOCALE_TIMEZONE]);
```
Comment (sjpadgett) [inline interface/modules/custom_modules/oe-module-comlink-telehealth/src/TelehealthGlobalConfig.php:455]: Yes I've seen this and meant to track it down.

File: interface/modules/custom_modules/oe-module-comlink-telehealth/moduleConfig.php [hunk 3]
```diff
@@ -37,11 +38,12 @@
...
-    $items['comlink_telehealth_user_id'] = $credentials['ctsiOrgUid'];
-    $items['comlink_telehealth_user_password'] = $cryptoGen->encryptStandard(trim($credentials['ctsiOrgPwd']));
-    $items['comlink_telehealth_cms_id'] = $credentials['ctsiOrgId'];
+    $items[TelehealthGlobalConfig::COMLINK_VIDEO_REGISTRATION_API] = $credentials['registrationUri'];
+    $items[TelehealthGlobalConfig::COMLINK_VIDEO_TELEHEALTH_API] = $credentials['videoApiUri'];
+    $items[TelehealthGlobalConfig::COMLINK_VIDEO_API_USER_ID] = $credentials['ctsiOrgUid'];
+    $items[TelehealthGlobalConfig::COMLINK_VIDEO_API_USER_PASSWORD] = $cryptoGen->encryptStandard(trim($credentials['ctsiOrgPwd']));
+    $items[TelehealthGlobalConfig::COMLINK_VIDEO_TELEHEALTH_CMS_ID] = $credentials['ctsiOrgId'];
+    $items[TelehealthGlobalConfig::COMLINK_TELEHEALTH_PAYMENT_SUBSCRIPTION_ID] = $credentials['paypal_subscription_id'] ?? null;
     // Save to globals table.
```
Comment (sjpadgett) [inline interface/modules/custom_modules/oe-module-comlink-telehealth/moduleConfig.php:47]: Recommend these default to empty string if undefined so exception won't get  thrown by sequel engine for writing null values to this table.
Or put a null key/value check in foreach.
Sorry I missed that in my commits.

I'm testing this on my test jig for pass through of new array key from server.
Will let ya know

Comment (adunsulag) [reply interface/modules/custom_modules/oe-module-comlink-telehealth/moduleConfig.php]: Fixed.

File: interface/modules/custom_modules/oe-module-comlink-telehealth/src/TelehealthGlobalConfig.php (hunk 2 above)
Comment (adunsulag) [reply interface/modules/custom_modules/oe-module-comlink-telehealth/src/TelehealthGlobalConfig.php:455]: I know where it's occurring, just thought that might have been intentional.  Didn't want to change anything in the globals though so close to the wire here so I just check against globals.

**Comment by adunsulag** [comment]
@sjpadgett if this is working well, we can bring this in and I'll submit a PR for 7.0.1

**Comment by sjpadgett** [comment]
My change to sign up application was just one line to the src/Action/Credentials/RequestCredentialAction.php on Sherwins git project. I also fix autocomplete to be off in credentials twig.
The change:
```
if ($this->data['credentials']) {
            $args = json_decode($newCredentials, true);
            self::saveCustomerCredentials(
                $this->repository,
                $this->customerValidator,
                $this->logerFactory
            );
            // So module can reference as needed.
            $args['ctsiVBData']['paypal_subscription_id'] = $this->data['paypal_subscription_id'];
            return $this->twig->render($response, 'displaycredentials.twig', $args);
        }
```

So with this PR and bypassing the paypal api request in my test jig it works.
I say @juggernautsei needs to update and test this in v7.0.1 we'll be releasing.

Go ahead and bring in if you're ready

**Review by sjpadgett — APPROVED** [review]
Great job especially I like the copy/paste you added.

**Comment by bradymiller** [comment]
2DV(1)

**Pull request merged**